src/app/shared/top-panel/user-panel/notifications-menu/notifications-menu.component.ts

```diff
@@ -0,0 +1,118 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {ChangeDetectionStrategy, Component, OnDestroy, OnInit, TemplateRef, ViewChild} from '@angular/core';
+import {UserNotification, UserNotificationType} from '../../../../core/model/user-notification';
+import {Observable, Subscription} from 'rxjs';
+import {select, Store} from '@ngrx/store';
+import {AppState} from '../../../../core/store/app.state';
+import {
+  selectAllUserNotifications,
+  selectUnreadUserNotifications,
+} from '../../../../core/store/user-notifications/user-notifications.state';
+import {UserNotificationsAction} from '../../../../core/store/user-notifications/user-notifications.action';
+import {UserNotificationsLoaderService} from '../../../../core/service/user-notifications-loader.service';
+import {OrganizationModel} from '../../../../core/store/organizations/organization.model';
+import {
+  selectAllOrganizations,
+  selectOrganizationsDictionary,
+} from '../../../../core/store/organizations/organizations.state';
+import {Dictionary} from '@ngrx/entity';
+import {Workspace} from '../../../../core/store/navigation/workspace.model';
+import {selectWorkspace} from '../../../../core/store/navigation/navigation.state';
+import {perspectiveIconsMap} from '../../../../view/perspectives/perspective';
+
+@Component({
+  selector: 'notifications-menu',
+  templateUrl: './notifications-menu.component.html',
+  styleUrls: ['./notifications-menu.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NotificationsMenuComponent implements OnInit {
+  public notifications$: Observable<UserNotification[]>;
+  public unreadNotifications$: Observable<UserNotification[]>;
+
+  public unreadOnly = false;
+
+  public perspectiveIcons = perspectiveIconsMap;
+
+  @ViewChild('organizationShared')
+  private organizationSharedTemplate: TemplateRef<any>;
+
+  @ViewChild('projectShared')
+  private projectSharedTemplate: TemplateRef<any>;
+
+  @ViewChild('collectionShared')
+  private collectionSharedTemplate: TemplateRef<any>;
+
+  @ViewChild('viewShared')
+  private viewSharedTemplate: TemplateRef<any>;
+
+  @ViewChild('unknown')
+  private unknownTemplate: TemplateRef<any>;
+
+  private organizations$: Observable<Dictionary<OrganizationModel>>;
+
+  private currentWorkspace$: Observable<Workspace>;
+
+  // need to include the notification loader service here for it to initially load notifications and to do that just once
+  constructor(private store: Store<AppState>, private notificationsLoader: UserNotificationsLoaderService) {}
+
+  public ngOnInit(): void {
```

Comment:
It is not necessary to use `void` return type.

Reply:
I agree with you but I think it is autogenerated like this by the IDE.